client/my-sites/stats/stats-module/all-time-nav.jsx

```diff
@@ -69,66 +72,82 @@ export const StatsModuleSummaryLinks = ( props ) => {
 		recordStats( item );
 	};
 
-	// Template for standard range options (7, 30, Quarter, Year, All Time).
-	const summaryPath = `/stats/day/${ path }/${ siteSlug }?startDate=${ moment().format(
-		'YYYY-MM-DD'
-	) }&summarize=1&num=`;
+	const getSummaryPath = useMemo( () => {
+		return ( numberDays ) => {
+			let url = window.location.href;
+			let queryParams = new URLSearchParams( url.split( '?' )[ 1 ] || '' );
 
-	// Path for summary or custom range option. ie: The first button in the row.
-	// Defaults to one day/week/month/year.
-	let summaryPeriodPath = `/stats/${
-		period.period
-	}/${ path }/${ siteSlug }?startDate=${ period.endOf.format( 'YYYY-MM-DD' ) }`;
-	// Override if custom range was used in query.
-	if ( query.start_date ) {
-		summaryPeriodPath = `/stats/${ period.period }/${ path }/${ siteSlug }?startDate=${ query.start_date }&endDate=${ query.date }`;
-	}
+			if ( isOdysseyStats ) {
+				url = window.location.href;
+				const hash = window.location.hash.split( '#!' )[ 1 ];
+				const hashQuery = new URLSearchParams( hash.split( '?' )[ 1 ] || '' );
+				queryParams = new URLSearchParams( hashQuery );
+			}
+
+			queryParams.set( 'startDate', moment().format( 'YYYY-MM-DD' ) );
```

Comment:
Is this an intentional change, as in the old logic it is:

```JavaScript
period.endOf.format( 'YYYY-MM-DD' )
```

Reply:
I was previously combining `summaryPath` and `summaryPathPeriod` [variables from the trunk previous version](https://github.com/Automattic/wp-calypso/blob/d7d5dade9d521f7aaf16108a83015b5d47b1af1f/client/my-sites/stats/stats-module/all-time-nav.jsx#L73-L85) into a single function and `summaryPath` used `moment().format`. I have now separated both the tasks so that this is clearer.